File: amf/master_crm/report/customer_satisfaction/customer_satisfaction.py
# Copyright (c) 2013, libracore AG and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _
from frappe.utils import flt

def execute(filters=None):
    """
    Returns two values:
      1) columns: list of dicts defining your report columns
      2) data: list of rows (each a dict matching fieldnames in columns)
    """
    filters = filters or {}
    from_date = filters.get("from_date")
    to_date = filters.get("to_date")

    # validation
    if not from_date or not to_date:
        frappe.throw(_("Please select both From Date and To Date"))
    if from_date > to_date:
        frappe.throw(_("From Date cannot be greater than To Date"))

    # fetch the average of amf_overall in the given date range
    sql = """
        SELECT
            AVG(amf_overall) AS global_average
        FROM `tabCustomer Satisfaction Survey`
        WHERE DATE(creation) BETWEEN %(from_date)s AND %(to_date)s
            AND docstatus = 1
    """
    result = frappe.db.sql(sql, {"from_date": from_date, "to_date": to_date}, as_dict=True)

    avg_value = flt(result[0].global_average*20 or 0, 2)

    # define columns
    columns = [
        {
            "fieldname": "global_average",
            "label": _("Global CSAT %"),
            "fieldtype": "Float",
            "width": 200
        }
    ]

    # single‐row result
    data = [
        {"global_average": avg_value}
    ]

    return columns, data

File: amf/master_crm/report/customer_satisfaction/customer_satisfaction.js
// Copyright (c) 2016, libracore AG and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Customer Satisfaction"] = {
    onload: function (report) {
        // today as YYYY-MM-DD
        const today = frappe.datetime.nowdate();
        const year = today.substr(0, 4);
        const start = `${year}-01-01`;
        const end = `${year}-12-31`;

        // set our two required filters
        report.set_filter_value("from_date", start);
        report.set_filter_value("to_date", end);
    },

    filters: [
        {
            fieldname: "from_date",
            label: "From Date",
            fieldtype: "Date",
            reqd: 1
        },
        {
            fieldname: "to_date",
            label: "To Date",
            fieldtype: "Date",
            reqd: 1
        }
    ]
};
